content-services/content-sources-backend
# Fixes 2017: add retry for failed tasks

## Summary
Adds a periodic requeue for delete-repository-snapshot tasks that have failed. Limited up to 3 retries.
Adds backoff timer for retries, with config option (tasking.retry_wait_upper_bound) for maximum wait time (wait time for final retry). 

## Testing steps
1. Add the `retry_wait_upper_bound` field to your config, with a low value like `1s`.
2. Create a snapshot
3. Get the delete-repository-snapshot task to fail. One way to do that is to do `podman stop cs_pulp_api_1`, so requests to the pulp server get a 502.
4. Observe that the task will get requeued and dequeued 3 times before permanently finishing in a failed state.

## Checklist
